### PR TITLE
applications: asset_tracker_v2: replace battery voltage with fuel gauge

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch uses the fuel gauge function introduced in PR 9526 to get a battery percentage instead of a voltage.

### CODE REVIEW

File: applications/asset_tracker_v2/src/events/sensor_module_event.h [hunk 1]
```diff
@@ -44,6 +44,11 @@ enum sensor_module_event_type {
 	 */
 	SENSOR_EVT_ENVIRONMENTAL_DATA_READY,
 
+	/** Battery fuel gauge data has been sampled.
+	 *  Payload is of type @ref sensor_module_data (batt).
```
Comment (jtguggedal) [inline applications/asset_tracker_v2/src/events/sensor_module_event.h:48]: `batt` -> `bat`

**Review by jtguggedal — APPROVED** [review]

**Comment by coderbyheart** [comment]
Could you provide an explanation in the commit, why percentage is better and voltage should be ignored?

**Comment by umapraseeda** [comment]
Do we need a changelog entry for this? If not, the label could be removed.

**Comment by jtguggedal** [comment]
> Do we need a changelog entry for this? If not, the label could be removed.

I think we need it

File: applications/asset_tracker_v2/src/modules/sensor_module.c [hunk 2]
```diff
@@ -242,6 +246,27 @@ static void apply_config(struct sensor_msg_data *msg)
 #endif /* CONFIG_EXTERNAL_SENSORS */
 }
 
+static void battery_data_get(void)
+{
+#if defined(CONFIG_ADP536X)
+	int rc;
```
Comment (simensrostad) [inline applications/asset_tracker_v2/src/modules/sensor_module.c:252]:
```suggestion
	int err;
```

File: applications/asset_tracker_v2/src/modules/sensor_module.c [hunk 3]
```diff
@@ -350,13 +387,17 @@ static void on_state_running(struct sensor_msg_data *msg)
...
+		    msg->module.app.count)) {
+			environmental_data_get();
 		}
 
-		environmental_data_get();
+		if (battery_data_requested(
+		    msg->module.app.data_list,
+		    msg->module.app.count)) {
+			battery_data_get();
+			}
```
Comment (simensrostad) [inline applications/asset_tracker_v2/src/modules/sensor_module.c:400]: Formatting shananigans?

**Review by simensrostad — COMMENTED** [review]
Should AWS, Azure, and LwM2M codecs be updated as well?

**Comment by maxd-nordic** [comment]
> Should AWS, Azure, and LwM2M codecs be updated as well?

As far as I could see only nRF Cloud mentioned the value as being a voltage, so I think it should work fine on all of them with the patch. Of course the range changes.

**Review by simensrostad — APPROVED** [review]

**Pull request merged**